app/v2/controllers/V2DeparturesController.scala

```diff
@@ -70,26 +75,56 @@ class V2DeparturesControllerImpl @Inject() (
   def submitDeclaration(): Action[Source[ByteString, _]] =
     (authActionNewEnrolmentOnly andThen messageSizeAction()).async(streamFromMemory) {
       implicit request =>
-        withTemporaryFile {
-          (temporaryFile, source) =>
-            logger.info("Version 2 of endpoint has been called")
-
-            implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
-
-            (for {
-              _ <- validationService.validateXML(MessageType.DepartureDeclaration, source).asPresentation
-
-              fileSource = FileIO.fromPath(temporaryFile)
-
-              declarationResult <- departuresService.saveDeclaration(request.eoriNumber, fileSource).asPresentation
-              _ <- routerService
-                .send(MessageType.DepartureDeclaration, request.eoriNumber, declarationResult.departureId, declarationResult.messageId, fileSource)
-                .asPresentation
-            } yield declarationResult).fold[Result](
-              presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
-              result => Accepted(HateoasDepartureDeclarationResponse(result.departureId))
-            )
-        }.toResult
+        request.headers.get("Content-Type") match {
+          case Some("application/xml")  => submitDeclarationXML
+          case Some("application/json") => submitDeclarationJSON
```

Comment:
I think play has constants that we should use here (I can't remember what the class is, I know there is `ContentTypes.JSON` and `ContentTypes.XML` - though the latter is a function and may not play well with pattern matching) 

That said, we also need to think about what we are going to instruct the traders to do. Are we going to enforce these two specific content types, or are we going to enable a range of things to be sent? Either way, we need to make it clear on the documentation what content types are accepted.

Reply:
import play.api.http.ContentTypes
import play.api.http.HeaderNames

Looks like XML is only a function in that it wraps mimetype in a charset, think it should be ok

Reply:
@drnaylor-oc @ardwalker I used play.api.http.MimeTypes

Reply:
Just to add, in clarification, we accepted we're only going to use `application/xml` and `application/json`. We should document this, and alongside that, document we only accept UTF-8 (as charsets can be specified in content types, and we want to explicity disallow that).